0000-catalog.md

```diff
@@ -0,0 +1,345 @@
+**catalog_id** ([UUID](https://crates.io/crates/uuid)), an **owner** (org_id 
+compatible w/Pike), an **expiry_date** (unix timestamp), a **Name**, and a 
+repeated field of **PropertyValues**.  The properties available are defined by 
+the Grid Property Schema Transaction Family and are restricted to the fields and 
+rules of the GS1 Catalog schema (which will be defined at a later time).  
+Transactions which are responsible for setting product state values must ensure 
+that those properties conform with the requirements of the GS1 Catalog Property 
+Schema (to be defined at a later time).
+
+``` 
+message Catalog { 
+    string catalog_id = 1; 
+    string owner = 2; 
+    string expiry_date = 3; 
+    name = 4; 
+    repeated PropertyValue properties = 6; 
+} 
+```
+
+### Referencing a Catalog
+
+Products are uniquely referenced by their product_id and product_namespace.  For
+GS1, Products are referenced by the GTIN identifier. For example:
+
+``` 
+get_catalog(catalog_id) // UUID 
+set_catalog(catalog_id, catalog) // UUID, catalog 
+```
+
+### Catalog Addressing in the Merkle-Radix State System
+
+In order to uniquely locate Catalogs in the Merkle-Radix state system, an
+address must be constructed which identifies the storage location of the Catalog
+representation.
+
+All Grid addresses are prefixed by the 6-hex-character namespace prefix
+"621dee",  Catalogs are further prefixed under the Grid namespace with reserved
+enumerations of "03" ("00", "01", and "02" being reserved for other purposes)
+indicating "Catalogs" and an additional "01" indicating "GS1 Catalog".
+
+Therefore, all addresses starting with:
+
+``` 
+"621dee" + "03" + "01" 
+```
+
+are Grid GS1 Catalogs identified by a UUID which can be reference by a 
+"catalog_product" to group products to a specific catalog. 
+
+UUID formats consist of 36-digit "alphanumeric strings" which include a fixed 
+amount of internal "0" padding.  After the 10-hex-characters that are consumed 
+by the grid namespace prefix, the catalog, and GS1 prefixes, there are 60 hex 
+characters remaining in the address.  The 36 digits of the GTIN can be left 
+padded with 24-hex-character zeroes and right padded with 2-hex-character zeroes 
+to accommodate potential future storage associated with the GS1 Catalog
+representation, for example:
+
+``` 
+"621dee" + "03" + "01" + 0000000000000000000000 +
+36-character "numeric string" catalog_id + "00" // catalog_id == UUID 
+```
+
+Using the v5 constructer in the UUID crate we can tie the catalog_id to 
+catalog_name will prevent duplication of catalogs when invoking the 
+catalog_replicate action.  
+```
+use uuid::Uuid;
+
+fn main() {
+    let my_uuid = Uuid::new_v5(&Uuid::new_v4(), b"catalog_name");
+    println!("{}", my_uuid);
+}
+
+>> 9db5fcec-dcab-5b04-b31b-765ccbf2bc3a
+```
+
+The full catalog address would look like:
+``` 
+"621dee030100000000000000000000009db5fcec-dcab-5b04-b31b-765ccbf2bc3a00" 
```

Comment:
Hyphens aren't a valid hexadecimal character so would this actually look like `621dee030100000000000000000000009db5fcecdcab5b04b31b765ccbf2bc3a00`?